src/OpenLoco/src/Vehicles/VehicleHead.cpp

```diff
@@ -179,12 +182,158 @@ namespace OpenLoco::Vehicles
         calculateRefundCost();
     }
 
+    // 0x004B8340
+    static void RecalculateTrainMinReliability(VehicleHead& head)
+    {
+        Vehicle train(head);
+        const auto getReliabilty = [](const Car& car) -> uint16_t {
+            if (car.front->reliability == 0)
+            {
+                return 0xFFFFU;
+            }
+            return car.front->reliability;
+        };
+        const auto minReliability = (*std::min_element(train.cars.begin(), train.cars.end(), [&getReliabilty](const Car& carLhs, const Car& carRhs) {
+                                        return getReliabilty(carLhs) < getReliabilty(carRhs);
+                                    })).front->reliability;
```

Comment:
Can we improve the formatting for this, somehow?

Reply:
just refactoring the predicate out makes it fairly nice

```const auto pred = [&getReliabilty](const Car& carLhs, const Car& carRhs) { return getReliabilty(carLhs) < getReliabilty(carRhs); };
const auto minReliability = (*std::min_element(train.cars.begin(), train.cars.end(), pred)).front->reliability;
```